native/src/widget/text_input/value.rs

```diff
@@ -58,9 +61,10 @@ impl Value {
     pub fn next_end_of_word(&self, index: usize) -> usize {
         let next_string = &self.graphemes[index..].concat();
 
-        UnicodeSegmentation::split_word_bound_indices(&next_string as &str)
-            .filter(|(_, word)| !word.trim_start().is_empty())
-            .next()
+        #[allow(clippy::or_fun_call)]
+        // self.len() is very cheap, we don't need lazy eval
```

Comment:
But why allocate memory every time it is not used?

Reply:
`self.len()` doesn't allocate any memory; it returns a usize which is stored on the stack. It will be stored on the stack even if we put it in a closure; so the closure is needless indirection.

Reply:
Same goes for here; Rust should inline the value.